.rubocop.yml

```diff
@@ -0,0 +1,29 @@
+AllCops:
+  TargetRubyVersion: 2.5
+  NewCops: disable
+  Include:
+    - 'lib/**/*'
```

Comment:
This needs to be the top-level `lib` only, not any of the `lib` subdirs in the current directory:

```
[2022/06/27 02:13:57.329] Running tests
[2022/06/27 02:13:57.329] Inspecting 144 files
[2022/06/27 02:13:57.329] ....................................................................................C................................................C..........
[2022/06/27 02:13:57.329] Offenses:
[2022/06/27 02:13:57.329] rubies/ruby-3.1.0/lib/ruby/gems/3.1.0/gems/ruby-prof-1.4.3/bin/ruby-prof:279:9: C: Security/Eval: The use of eval is a serious security risk.
[2022/06/27 02:13:57.329]         eval(exec)
[2022/06/27 02:13:57.329]         ^^^^
[2022/06/27 02:13:57.329] rubies/ruby-3.1/lib/ruby/gems/3.1.0/gems/ruby-prof-1.4.3/bin/ruby-prof:279:9: C: Security/Eval: The use of eval is a serious security risk.
[2022/06/27 02:13:57.329]         eval(exec)
[2022/06/27 02:13:57.329]         ^^^^
[2022/06/27 02:13:57.329] 144 files inspected, 2 offenses detected
```

Reply:
Hmm... this failure looks Evergreen specific. You can see the Github actions build is passing here: https://github.com/mongodb/bson-ruby/runs/7051219018?check_suite_focus=true

Since I can't access Evergreen, I think you're going to have to debug this. Possibly Evergreen is not running Rubocop from the project root dir. There are a number of things to try, some ideas:
- check Rubocop is run from project root
- try `Include:` arg as `/path/from/systemroot/bson-ruby/lib/**/*`
etc.

Reply:
We have the Ruby installation together with gems under the project root in evergreen.

> The syntax I'm using here Include: - lib/**/* is standard as per docs and means "from project root".

`lib/**/*` in shell means `./lib/**/*`. In Rubocop this apparently means `./**/lib/**/*`. After a quick look at the configuration page I couldn't find where `Include` is described, but the configuration in this PR needs to match the shell behavior of the glob as I just described it.

Reply:
Again, Rubocop is working correctly on my local machine, and the specs are passing on GH Actions.

I have no way of seeing and knowing what Evergreen is doing.

Reply:
As a shot in the dark, let's try `/lib/**/*` instead of `lib/**/*`

Reply:
Please find out whether `/lib/**/*` actually traverses from filesystem root and report your findings.

Reply:
@p-mongo I have done this already. Here's the evidence:

```
/workspace/rubocop-test$ ruby -e "puts Dir.glob('**/*')"
foo
foo/lib
foo/lib/test.rb
foo/lib/test1
foo/lib/test1/test.rb
lib
lib/test.rb
lib/test1
lib/test1/test.rb

# this is using the exact rubocop.yml in this PR
/workspace/rubocop-test$ rubocop -L
lib/test.rb
lib/test1/test.rb

# note there is also a /lib dir in my drive root
```

Please kindly run Evergreen as a next step, thanks!

Reply:
This is the same line as it was before, no?

Reply:
Yes. As I have stated, this line `lib/**/*` is working correctly on:
1. bson-ruby repo on my local machine
2. the test case above, which demonstrates it is only reading `lib` in root dir under normal circumstances.
3. in Github Actions, see green checkmark below

The only thing that is failing is Evergreen. In order to debug this further, running `rubocop -L` will provide a clear indication of where Rubocop is actually scanning on Evergreen, so I can fix it.

(You are making the assumption that `Include: "lib/**/*"` is causing Rubocop to scan lib folders in nested directories, however, it may just be coincidental, because most Ruby files are found in `lib` dirs somewhere.)